engineapi/proxy.go

```diff
@@ -100,69 +323,78 @@ func (h *EngineClientProxyHandler) GetClient(im *longhorn.InstanceManager) (c En
 		return nil, errors.Errorf("BUG: failed to get proxy client due to missing instance manager")
 	}
 
-	log := getLoggerForProxy(h.logger, im.Spec.Image)
+	_, exist := h.GetHandledInstanceManagerImage(im.Spec.Image)
+	if !exist {
+		for id := 0; id < DefaultProxyClientPerImage; id++ {
+			h.NewHandleClient(im, id)
+		}
+	}
 
-	h.lock.Lock()
-	client, exist := h.Clients[im.Spec.Image]
-	h.lock.Unlock()
-	if exist {
-		err := client.Ping()
-		if err == nil {
-			return client, nil
+	var candidate EngineClientProxy
+	failCount := 0
+	for {
+		if failCount > ProxyClientMaxRetry {
```

Comment:
This is tricky. I know you want to make GetClient still work as blocking to ensure it can get the correct in a best effort.